src/metax_api/models/catalog_record.py

```diff
@@ -312,9 +313,10 @@ def user_is_owner(self, request):
 
     def save(self, *args, **kwargs):
         if self._operation_is_create():
-            self._pre_create_operations()
+            pref_id_type = self._get_preferred_identifier_type_from_request()
```

Comment:
Instead of passing around as a parameter, I would recommend leaving the save() method as it was, and instead extract the pid_type for the first time inside pre_create_operations(),and in post_create_operations() checking the identifier type using get_identifier_type().